src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderCursor.cs

```diff
@@ -417,15 +420,23 @@ private sealed class LineReader
                 private Task _thdRead;
                 private volatile bool _abort;
 
-                public LineReader(IMultiStreamSource files, int batchSize, int bufSize, bool hasHeader, bool readMultilines, char[] separators, char escapeChar, long limit, int cref)
+                // The channel to report messages on.
+                private readonly IChannel _ch;
+                // text file header string if any
+                private string _headerString;
+
+                public LineReader(IChannelProvider provider, IMultiStreamSource files, int batchSize, int bufSize,
+                    bool hasHeader, bool readMultilines, char[] separators, char escapeChar, long limit, int cref)
```

Comment:
**Question:** Just curious, why did you change the IHostEnvironment parameter you had here on a previous commit to an IChannelProvider? Was this related to the failures this PR had on the CI?